### DISCUSSION TITLE
Cameraview - Determining which Camera Infos are wide angle

### DISCUSSION THREAD

**Discussion by kidpresto92** [CommunityToolkit/Maui#2347] [General]
We're trying to transition from the CameraView in an older Maui camera library, https://github.com/hjam40/Camera.MAUI over to the MCT one.

We've noticed that there isn't a straightforward way of determining which CameraInfos provided are wide angle lens or regular angle lens. 

With hjam40's camera library, we were able to determine which lens was the wide angle lens based on the horizontal viewing angles, 

- [Apple/MauiCameraView.cs](https://github.com/hjam40/Camera.MAUI/blob/bc9dcb08009633a2fdff37d75494374d988a3369/Camera.MAUI/Apple/MauiCameraView.cs#L97)
- [Android/MauiCameraView.cs](https://github.com/hjam40/Camera.MAUI/blob/bc9dcb08009633a2fdff37d75494374d988a3369/Camera.MAUI/Platforms/Android/MauiCameraView.cs#L111C21-L111C108)

With the camera info that we receive from the Maui Community Toolkit, it looks like the only way we could determine this would be the names of the cameras. Is there a better way to determine this? 

Examples of camera info received from a few different devices with the MCT:
[MCT camera info examples.json](https://github.com/user-attachments/files/17778197/MCT.camera.info.examples.json)

**Comment by bijington** [comment]
At present there is no way to determine this. We would happily accept this functionality though if you are willing to provide it?

**Reply by kidpresto92** [reply]
I'll see if I can get my product owners to devote time to it. But I wouldn't count on it🙃. Thanks for the quick response!

**Comment by dhindrik** [comment]
My solution right now is that I added the device ids to the code. I think they are mapped to AVCaptureDevice.DeviceType, https://developer.apple.com/documentation/avfoundation/avcapturedevice/devicetype-swift.struct, so it should be fun. 

But for Android I only get two cameras, one for rear and one for front. So does that mean that wide angle is not supported for Android? Camera app has 0.6 as an zoom option, but MinimunZoom starts on 1 for Android so it is not possible to do it by setting a low zoom value either.
